SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Channel/Channel.cs

```diff
@@ -336,158 +298,132 @@ private Task Suspend(bool deleteLogs, Exception exception)
-                    _mutex.Unlock();
+                    SentLog?.Invoke(this, new SentLogEventArgs(log));
                 }
-            };
-            serviceCall.Execute();
+            }
         }
 
-        private void HandleSendingFailure(string batchId, IngestionException e)
+        private async Task HandleSendingFailureAsync(State state, string batchId, IngestionException e)
         {
             var isRecoverable = e?.IsRecoverable ?? false;
             MobileCenterLog.Error(MobileCenterLog.LogTag, $"Sending logs for channel '{Name}', batch '{batchId}' failed", e);
-            var removedLogs = _sendingBatches[batchId];
-            _sendingBatches.Remove(batchId);
-            if (!isRecoverable && FailedToSendLog != null)
-            {
-                foreach (var log in removedLogs)
-                {
-                    FailedToSendLog?.Invoke(this, new FailedToSendLogEventArgs(log, e));
-                }
-            }
-            _mutex.Lock();
-            try
+            List<Log> removedLogs;
+            using (await _mutex.GetLockAsync(state).ConfigureAwait(false))
             {
-                Suspend(!isRecoverable, e);
+                removedLogs = _sendingBatches[batchId];
+                _sendingBatches.Remove(batchId);
                 if (isRecoverable)
                 {
                     _pendingLogCount += removedLogs.Count;
                 }
             }
-            finally
+            if (!isRecoverable && FailedToSendLog != null)
             {
-                _mutex.Unlock();
+                foreach (var log in removedLogs)
+                {
+                    FailedToSendLog?.Invoke(this, new FailedToSendLogEventArgs(log, e));
+                }
             }
+            await SuspendAsync(state, !isRecoverable, e).ConfigureAwait(false);
         }
 
-        private void CheckPendingLogs()
+        private async Task CheckPendingLogsAsync(State state)
         {
             if (!_enabled)
             {
                 MobileCenterLog.Info(MobileCenterLog.LogTag, "The service has been disabled. Stop processing logs.");
-                return;
             }
 
             MobileCenterLog.Debug(MobileCenterLog.LogTag, $"CheckPendingLogs({Name}) pending log count: {_pendingLogCount}");
             if (_pendingLogCount >= _maxLogsPerBatch)
             {
-                Task.Run(TriggerIngestionAsync);
+                await TriggerIngestionAsync(state).ConfigureAwait(false);
             }
             else if (_pendingLogCount > 0 && !_batchScheduled)
             {
                 _batchScheduled = true;
-                Task.Delay((int)_batchTimeInterval.TotalMilliseconds).ContinueWith(async completedTask =>
+
+                // No need wait _batchTimeInterval here.
+#pragma warning disable CS4014
+                Task.Run(async () =>
                 {
+                    await Task.Delay((int)_batchTimeInterval.TotalMilliseconds).ConfigureAwait(false);
```

Comment:
Instead of wrapping this in a `Task.Run`, why not use `ContinueWith`? You don't need to `await` the result